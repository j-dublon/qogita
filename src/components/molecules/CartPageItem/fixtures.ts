import { CartPageItemProps } from "@/types";

export const mockHandleUpdateCart = jest.fn();

export const mockProps: CartPageItemProps = {
  item: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    price: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    id: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
    quantity: 3,
  },
  handleUpdateCart: mockHandleUpdateCart,
};
