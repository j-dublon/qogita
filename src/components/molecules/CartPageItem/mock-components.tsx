import { CartQuantityButtonsProps } from "@/types";
import React, { FC } from "react";

export const MockUpdateCartQuantityButtons: FC<CartQuantityButtonsProps> = ({
  updateCartQuantity,
}) => (
  <div onClick={() => updateCartQuantity(5)}>Update Cart Quantity Buttons</div>
);
