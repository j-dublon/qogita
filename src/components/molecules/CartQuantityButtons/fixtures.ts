import { CartQuantityButtonsProps } from "@/types";

export const mockUpdateCartQuantity = jest.fn();

export const mockProps: CartQuantityButtonsProps = {
  product: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
  },
  numberInCart: 3,
  updateCartQuantity: mockUpdateCartQuantity,
};
