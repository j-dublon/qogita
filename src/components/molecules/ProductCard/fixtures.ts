import { Product } from "@/types";

export const mockProps: Product = {
  name: "Mock Product",
  imageUrl: "http://your-product.com",
  recommendedRetailPrice: 52,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "987",
  brandName: "Chanel",
  categoryName: "Perfume",
};
