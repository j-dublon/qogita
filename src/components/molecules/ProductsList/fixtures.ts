import { ProductsListProps } from "@/types";

export const mockProps: ProductsListProps = {
  products: [
    {
      name: "Korres Endocare C Proteoglicanos Spf30 Ampoules 30x2ml",
      gtin: "8470001901682",
      recommendedRetailPrice: 59.99,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/cyW4Zm7C5EDgmWYLkGVQ6U.jpg",
      brandName: "Endocare",
      categoryName: "Women's Perfume",
    },
    {
      name: "Indasec Instituto Espa√±ol Argan Amphora Oil 400ml",
      gtin: "8411047133156",
      recommendedRetailPrice: 6.45,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/GPTKSirQGkJEKiS4PvAqtJ.jpg",
      brandName: "Indasec",
      categoryName: "Sundry",
    },
    {
      name: "Schwarzkopf 3d Mension Molding Wax - 100 Ml - Wax",
      gtin: "4045787263770",
      recommendedRetailPrice: 14.4,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/mv4rECdJqtdg2SnLM66LN8.jpg",
      brandName: "Schwarzkopf Professional",
      categoryName: "Hair Wax",
    },
    {
      name: "Veet Cold Wax Strips Armpits",
      gtin: "3059944026321",
      recommendedRetailPrice: 17.49,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/RoYyVChLGF94mZFtMTAHdi.jpg",
      brandName: "Veet",
      categoryName: "Depilatory Creams, Wax & Resin",
    },
    {
      name: "Philip B - Katira Hair Masque 178 Ml",
      gtin: "0893239000824",
      recommendedRetailPrice: 55.49,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/6dSNi2b8DsmaSRoNrDDNYX.jpg",
      brandName: "Philip B",
      categoryName: "Women's Perfume",
    },
    {
      name: "Electrolux Ekf7900 Counter Top Filter Coffee Maker 1.65 L Fully Automatic",
      gtin: "7332543331482",
      recommendedRetailPrice: 129.95,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/7eiCzQroBcf8NPLiqPWBN2.jpg",
      brandName: "Electrolux",
      categoryName: "Filter Coffee Makers",
    },
  ],
};
