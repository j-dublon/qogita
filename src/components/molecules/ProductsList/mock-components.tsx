import React, { FC } from "react";
import { Product } from "@/types";

export const MockProductCard: FC<Product> = ({ name }) => <div>{name}</div>;
