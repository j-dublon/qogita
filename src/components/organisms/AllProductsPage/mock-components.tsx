import { AllProductsPageProps } from "@/types";
import React, { FC } from "react";

export const MockLoading: FC = () => <div>Loading...</div>;

export const MockError: FC = () => <div>Error</div>;

export const MockProductsList: FC = () => <div>Products List</div>;

export const MockPagination: FC = () => <div>Pagination</div>;

export const MockAllProductsPage: FC<AllProductsPageProps> = ({
  fetchData,
}) => (
  <div>
    Products Page
    <div onClick={() => fetchData(1)}>Products Page Pagination</div>
  </div>
);
