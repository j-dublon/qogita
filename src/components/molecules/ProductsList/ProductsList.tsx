import { FC } from "react";
import { ProductsListProps } from "../../../types";
import ProductCard from "../ProductCard/ProductCard";

const ProductsList: FC<ProductsListProps> = ({ products }) => (
  <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-4 mb-8">
    {products.map((product) => (
      <ProductCard {...product} />
    ))}
  </div>
);

export default ProductsList;
