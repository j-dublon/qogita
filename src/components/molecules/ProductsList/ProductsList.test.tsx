import React from "react";
import { render } from "@testing-library/react";
import ProductsList from "./ProductsList";
import { mockProps } from "./fixtures";
import { Product } from "@/types";
import { MockProductCard } from "./mock-components";

jest.mock("../ProductCard/ProductCard", () => ({
  __esModule: true,
  default: (props: Product) => <MockProductCard {...props} />,
}));

describe("Component: ProductsList", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<ProductsList {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it.each(mockProps.products)(
    `SHOULD render a ProductCard component for each product WHEN props are provided`,
    (product) => {
      const { getByText } = render(<ProductsList {...mockProps} />);
      expect(getByText(product.name));
    }
  );
});
