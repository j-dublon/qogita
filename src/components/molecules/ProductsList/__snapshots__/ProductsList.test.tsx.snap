// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductsList SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-4 mb-8"
        role="list"
      >
        <div>
          Korres Endocare C Proteoglicanos Spf30 Ampoules 30x2ml
        </div>
        <div>
          Indasec Instituto Español Argan Amphora Oil 400ml
        </div>
        <div>
          Schwarzkopf 3d Mension Molding Wax - 100 Ml - Wax
        </div>
        <div>
          Veet Cold Wax Strips Armpits
        </div>
        <div>
          Philip B - Katira Hair Masque 178 Ml
        </div>
        <div>
          Electrolux Ekf7900 Counter Top Filter Coffee Maker 1.65 L Fully Automatic
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-4 mb-8"
      role="list"
    >
      <div>
        Korres Endocare C Proteoglicanos Spf30 Ampoules 30x2ml
      </div>
      <div>
        Indasec Instituto Español Argan Amphora Oil 400ml
      </div>
      <div>
        Schwarzkopf 3d Mension Molding Wax - 100 Ml - Wax
      </div>
      <div>
        Veet Cold Wax Strips Armpits
      </div>
      <div>
        Philip B - Katira Hair Masque 178 Ml
      </div>
      <div>
        Electrolux Ekf7900 Counter Top Filter Coffee Maker 1.65 L Fully Automatic
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
